core/vm/vmcontext/context.go

```diff
@@ -142,25 +142,30 @@ func VerifySealFn(ref *types.Header, chain chainContext) func(*types.Header) boo
 // If the calculation or transfer of the tax amount fails for any reason, the regular transfer goes ahead.
 // NB: Gas is not charged or accounted for this calculation.
 func TobinTransfer(evm *vm.EVM, sender, recipient common.Address, amount *big.Int) {
-	// Run only primary evm.Call() with tracer
-	if evm.GetDebug() {
-		evm.SetDebug(false)
-		defer func() { evm.SetDebug(true) }()
-	}
+	// Only deduct tobin tax before the g hardfork
+	if evm.ChainConfig().IsGFork(evm.Context.BlockNumber) {
+		Transfer(evm.StateDB, sender, recipient, amount)
+	} else {
+		// Run only primary evm.Call() with tracer
+		if evm.GetDebug() {
+			evm.SetDebug(false)
+			defer func() { evm.SetDebug(true) }()
+		}
 
-	if amount.Cmp(big.NewInt(0)) != 0 {
```

Comment:
You could reduce this change to a one line diff with
```
	if amount.Cmp(big.NewInt(0)) != 0 && !evm.ChainConfig().IsGFork(evm.Context.BlockNumber) {
```
I'm not sure if that is clearer, though (and it would keep flipping the debugging off and on without need).

Reply:
I like the clear separation between logic and hardfork, so I'd prefer to keep it as is.